lib/push.py

```diff
@@ -81,24 +89,75 @@ def manifest_new(class_):
                "weirdal": "yankovic" }
 
    def cleanup(self):
+      # Delete the config, manifest, and tarballs if the cache is disabled.
+      conf_suffix = str(self.image) + ".config.json"
+      conf_path = ch.storage.upload_cache // conf_suffix
+      man_suffix = str(self.image) + ".manifest.json"
+      man_path = ch.storage.upload_cache // man_suffix
+      if (not conf_path.exists() or not man_path.exists()):
+         return
```

Comment:
```suggestion
      # Delete uncached tarballs.
```
Push should behave exactly like it does now when not using `--cache-upload`. Other files are left alone and written over as needed.